File: node/src/poc.ts
import express from "express";
import { ListContinents } from "./continent/continent";
import { GetImage } from "./continent/imageHandler";

const app = express();

app.listen(3000, () => console.log("Ready"));

app.get("/continent", (req, res) => {
    res.send(ListContinents(""));
});

app.get("/continent/:name", (req, res) => {
    const continent = ListContinents(req.params.name);
    if (!continent) {
        res.status(404).send("Could not find continent with desired name.");
    } else {
        res.send(continent);
    }
});

app.get("/continent/image/:name", (req, res) => {
    const image = GetImage(req.params.name);
    if (!image) {
        res.status(404).send("Could not find image with desired name.");
    } else {
        res.type("image/svg+xml");
        res.send(image);
    }
});

File: go/native/src/poc.go
package main

import (
	"encoding/json"
	"log"
	"net/http"

	"github.com/getsalty/Webservice_POC/go/native/src/continent"
)

func main() {
	http.HandleFunc("/continent", func(w http.ResponseWriter, r *http.Request) {
		w.Header().Add("Content-Type", "application/json")

		var continentList []continent.Continent
		continent.ListContinents(r.URL.Path, &continentList)
		if continentList != nil {
			json.NewEncoder(w).Encode(continentList)
		}
	})

	http.HandleFunc("/continent/", func(w http.ResponseWriter, r *http.Request) {
		var continentObject []continent.Continent
		continent.ListContinents(r.URL.Path, &continentObject)

		if continentObject == nil {
			w.WriteHeader(404)
			w.Write([]byte(`Could not find continent with desired name.`))
		} else {
			w.Header().Add("Content-Type", "application/json")
			if len(continentObject) == 1 {
				json.NewEncoder(w).Encode(continentObject[0])
			} else {
				json.NewEncoder(w).Encode(continentObject)
			}
		}

	})

	http.HandleFunc("/continent/image/", func(w http.ResponseWriter, r *http.Request) {
		image := continent.GetImage(r.URL.Path)

		if image == nil {
			w.WriteHeader(404)
			w.Write([]byte(`Could not find image with desired name.`))
		} else {
			w.Header().Add("Content-Type", "image/svg+xml")
			w.Write(image)
		}
	})

	log.Fatal(http.ListenAndServe(":4000", nil))
}
